core/src/test/java/org/testcontainers/junit/GenericContainerRuleTest.java

```diff
@@ -127,6 +128,15 @@ public static void setupContent() throws FileNotFoundException {
             .withExtraHost("somehost", "192.168.1.10")
             .withCommand("/bin/sh", "-c", "while true; do cat /etc/hosts | nc -l -p 80; done");
 
+    /**
+     * Create a container with a custom label for testing.
+     */
+    @ClassRule
+    public static GenericContainer alpineCustomLabel = new GenericContainer("alpine:3.2")
```

Comment:
please use `try-with-resources`, otherwise this container will be started for **any** test of this clss

Reply:
Also, the container for this test can be as simple as:
```java
new GenericContainer()
    .withLabel("our.custom", "label")
    .withCommand("ps")
```

Reply:
So it behaves like every other container in this test class ;)

Reply:
not every, rather legacy ;)

Reply:
See `createContainerCmdHookTest`, `copyToContainerTest` and others

Reply:
I see - much better :)